app/src/components/login-form/login-form.ts

```diff
@@ -0,0 +1,45 @@
+import {Component, Input, Output, EventEmitter} from 'angular2/core';
+import {Router} from 'angular2/router';
+import {
+  FORM_BINDINGS,
+  FORM_DIRECTIVES,
+  ControlGroup,
+  FormBuilder,
+  Validators
+} from 'angular2/common';
+import AuthService from '../../services/auth-service';
+const TEMPLATE = require('./login-form.html');
+
+declare interface ILoginRequest {
+  username: string,
+  password: string
+};
+
+@Component({
+  selector: 'login-form',
+  directives: [FORM_DIRECTIVES],
+  viewBindings: [FORM_BINDINGS],
+  template: TEMPLATE
+})
+export default class LoginForm {
+
+  @Input() message: string = '';
```

Comment:
thinking we should switch to `@Input` and `@Output` to be consistent - going to be a bit of an ongoing refactor with that. Going to try and keep those commits cherry-pickable so I can apply them to a few branches easily.